in-person-payments-example/Controllers/WebhookController.cs

```diff
@@ -0,0 +1,183 @@
+using Adyen.Model.Notification;
+using Adyen.Util;
+using adyen_dotnet_in_person_payments_example.Models;
+using adyen_dotnet_in_person_payments_example.Options;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using adyen_dotnet_in_person_payments_example.Services;
+
+namespace adyen_dotnet_in_person_payments_example.Controllers
+{
+    [ApiController]
+    public class WebhookController : ControllerBase
+    {
+        private readonly ILogger<WebhookController> _logger;
+        private readonly ITableService _tableService;
+        private readonly HmacValidator _hmacValidator;
+        private readonly string _hmacKey;
+
+        public WebhookController(ILogger<WebhookController> logger, IOptions<AdyenOptions> options, ITableService tableService, HmacValidator hmacValidator)
+        {
+            _logger = logger;
+            _tableService = tableService;
+            _hmacValidator = hmacValidator;
+            _hmacKey = options.Value.ADYEN_HMAC_KEY;
+        }
+
+        [HttpPost("api/webhooks/notifications")]
+        public async Task<ActionResult<string>> Webhooks(NotificationRequest notificationRequest)
+        {
+            _logger.LogInformation($"Webhook received::\n{notificationRequest.ToJson()}");
+
+            try
+            {
+                // JSON and HTTP POST notifications always contain a single `NotificationRequestItem` object.
+                // Read more: https://docs.adyen.com/development-resources/webhooks/understand-notifications#notification-structure.
+                NotificationRequestItemContainer container = notificationRequest.NotificationItemContainers?.FirstOrDefault();
+
+                if (container == null)
+                {
+                    return BadRequest("Container has no notification items");
+                }
+
+                // We always recommend to activate HMAC validation in the webhooks for security reasons.
+                // Read more here: https://docs.adyen.com/development-resources/webhooks/verify-hmac-signatures & https://docs.adyen.com/development-resources/webhooks#accept-notifications
+                if (!_hmacValidator.IsValidHmac(container.NotificationItem, _hmacKey))
+                {
+                    _logger.LogError($"Error while validating HMAC Key");
+                    return BadRequest("[not accepted invalid hmac key]");
+                }
+
+                // Process CANCEL_OR_REFUND notification asynchronously (also known as: `reversals`).
+                // Documentation: https://docs.adyen.com/point-of-sale/basic-tapi-integration/refund-payment/refund-webhooks/#cancel-or-refund-webhook
+                await ProcessCancelOrRefundNotificationAsync(container.NotificationItem);
```

Comment:
should it also process the `authorisation` event?

Reply:
I've added the AUTHORISATION webhook processing, we recommend it because the merchant probably want to persist some data in their own backend and perform some business logic on it on their side.